packages/core/src/repository/query/network.query.ts

```diff
@@ -0,0 +1,93 @@
+import { ParameterType } from '../../data';
+import {KeyQuery} from "./query";
+
+export enum HttpMethod {
+    Get = 'GET',
+    Post = 'POST',
+    Put = 'PUT',
+    Delete = 'DELETE',
+}
+
+export abstract class NetworkQuery extends KeyQuery {
+    constructor(key = '') {
```

Comment:
It's OK to default to `''`?? I understand we don't want to make this required. At the same time if we have the cache set and we forget to set the `key`… it can be confusing.

Does it make sense to create a random key instead? We have the `randomString` utility… This probably can also be confusing.

Maybe we should log a warning on the `CacheRepository` if the `key = ''`???